### DISCUSSION TITLE
treeview in Xreader opens in random state

### DISCUSSION THREAD

**Discussion by CignaD** [linuxmint/discussions#711] [Ideas]
Issue: when a pdf file containing table-of-contents style bookmarks is open in Xreader, the tree view seems to be in a random state with some branches collapsed and some expanded.

Suggestions: 

1. When a file is opened for the first time, all branches in the tree should be either collapsed or expanded, preferably according to an option in preferences.
2. There should be "collapse all" and "expand all" buttons at the top of the pane when tree style bookmarks are displayed.
3. (nice to have) When a pdf file is closed, the state of the treeview should be saved along with the page currently being viewed and restored when the file is reopened.

I looked at the source code with the intention making the changes myself, but I have nowhere near the level of competence required.

**Comment by JackNWhite** [comment]
I have submitted [a change](https://github.com/linuxmint/xreader/pull/672) to simply make the collapsed state the default. Sorry it's not more than that, and I don't know whether it will be approved.
